torch/_inductor/ir.py

```diff
@@ -3627,7 +3628,10 @@ def __init__(self, name, input_nodes, layout):
 
     def benchmark(self, *args, out) -> float:
         algo = self.to_callable()
-        return do_bench(lambda: algo(*args, out=out))
+        if is_cpu_device(args):
+            return do_bench_cpu(lambda: algo(*args, out=out))
+        else:
+            return do_bench(lambda: algo(*args, out=out))
```

Comment:
Can we have this device dispatch logic inside do_bench?

Basically, rename current do_bench to do_bench_gpu. And create a new do_bench function that dispatches to do_bench_cpu/do_bench_gpu based on devices.

Reply:
Thanks. I did the refactoring in a separate PR: https://github.com/pytorch/pytorch/pull/125736